src/coreclr/vm/appdomain.cpp

```diff
@@ -2778,7 +2778,7 @@ void AppDomain::SetFriendlyName(LPCWSTR pwzFriendlyName)
     CONTRACTL
     {
         THROWS;
-        if (GetThreadNULLOk()) {GC_TRIGGERS;} else {DISABLED(GC_NOTRIGGER);}
+        if (GetThreadNULLOk()) {DISABLED(GC_NOTRIGGER);} else {GC_TRIGGERS;}
```

Comment:
```suggestion
        GC_NOTRIGGER;
```
The only place that it is called from has thread setup.